apps/sms/test/marionette/lib/views/activity-chooser/view.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+/* global module */
+
+var MenuAccessor = require('../shared/menu_accessors');
+
+function ActivityChooserView(client) {
+  this.client = client;
+  this.menuAccessors = new MenuAccessor(client);
+}
+
+ActivityChooserView.prototype = {
+  choose: function(appName) {
+    this.menuAccessors.selectSystemMenuOption(appName);
+
+    var activityApp = this.createAppViewByName(appName);
+    activityApp.switchTo();
+
+    return activityApp;
+  },
+
+  createAppViewByName: function(appName) {
+    switch(appName) {
```

Comment:
to answer your previous question, I'd say "why not"; this is like a factory, and factories usually have such switch/case instructions.